### PR TITLE
Update bbp-morphology-workflow to use NeuroR

### PR DESCRIPTION
**Pull request by wizmer**
which merges morph-repair and cut-plane (that's why cut-plane folder is being deleted)

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-neuror/package.py [hunk 1]
```diff
@@ -7,17 +7,18 @@
 from spack import *
 
 
-class PyMorphRepair(PythonPackage):
+class PyNeuror(PythonPackage):
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-neuror/package.py:10]: I was somehow expecting `NeuroR` to be some `R` bindings…

I think renaming stuff like this is somewhat dangerous. Currently installed packages depending on `py-morph-repair` will be missing the package file…

**Comment by wizmer** [comment]
Hi, I don't understand what is wrong with this build. Searching for 'failure' or 'error' does not find anything in the log. Do you know which part of the CI failed ?

The only thing I have is:
```
10:45:13  ==> Warning: Could not write module file [/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/deploy/applications/2020-03-13/modules/tcl/linux-rhel7-x86_64/py-bbp-morphology-workflow/1.0.1]
10:45:13  ==> Warning: 	--> Package 'py-morph-repair' not found. <--
```
py-morph-repair has been removed on purpose but I don't know why the module file of py-bbp-morphology-workflow is not being written.

I have also tried to run the command as suggested in the FAQ:
```
./share/spack/qa/run-unit-tests
```
But it complains about subversion:
```
bcoste@bbpv1:~/workspace/spack$ ./share/spack/qa/run-unit-tests
ERROR: svn is required to run this script.

To install with Spack, run:
    $ spack install subversion
Then add the bin directory to your PATH.
```
I have installed subversion via `spack install` but I cannot load it:
```
bcoste@bbpv1:~/workspace/spack$ spack load subversion
ModuleCmd_Load.c(213):ERROR:105: Unable to locate a modulefile for 'subversion/1.12.2'
```
Could I have some help to investigate this ? 
Thanks a lot

File: var/spack/repos/builtin/packages/py-neuror/package.py (hunk 1 above)
Comment (wizmer) [reply var/spack/repos/builtin/packages/py-neuror/package.py:10]: I wish we had kept the old name but unfortunately the rename is a decision of the open-sourcing committee.

**Comment by matz-e** [comment]
> Hi, I don't understand what is wrong with this build. Searching for 'failure' or 'error' does not find anything in the log. Do you know which part of the CI failed ?

The best way to consume Jenkins here is BleuOcean. Click on the tests tab there, and you will see which packages failed to build.

The error I see there is:
```
Traceback (most recent call last):
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/report.py", line 137, in wrapper
value = do_install(pkg, *args, **kwargs)
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/package.py", line 1590, in do_install
partial = self.check_for_unfinished_installation(keep_prefix, restage)
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/package.py", line 1860, in check_for_unfinished_installation
if restage and self.stage.managed_by_spack:
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/package.py", line 806, in stage
self._stage = self._make_stage()
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/package.py", line 783, in _make_stage
stage = self._make_root_stage(fetcher)
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/package.py", line 761, in _make_root_stage
self.spec)
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/mirror.py", line 313, in mirror_archive_paths
ext = ext or _determine_extension(fetcher)
File "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/748/spack/lib/spack/spack/mirror.py", line 260, in _determine_extension
raise MirrorError(msg.format(fetcher.url))
spack.mirror.MirrorError: Unable to parse extension from ssh://bbpcode.epfl.ch/platform/bbp-morphology-workflow.
If this URL is for a tarball but does not include the file extension
in the name, you can explicitly declare it with the following syntax:
version('1.2.3', 'hash', extension='tar.gz')
If this URL is for a download like a .jar or .whl that does not need
to be expanded, or an uncompressed installation script, you can tell
Spack not to expand it with the following syntax:
version('1.2.3', 'hash', expand=False)
```

> py-morph-repair has been removed on purpose but I don't know why the module file of py-bbp-morphology-workflow is not being written.

See my comment above about this. I'll have to research what upstream Spack does for renaming.

> I have also tried to run the command as suggested in the FAQ:
> 
> ```
> ./share/spack/qa/run-unit-tests
> ```
> 
> But it complains about subversion:
> 
> ```
> bcoste@bbpv1:~/workspace/spack$ ./share/spack/qa/run-unit-tests
> ERROR: svn is required to run this script.
> 
> To install with Spack, run:
>     $ spack install subversion
> Then add the bin directory to your PATH.
> ```
> 
> I have installed subversion via `spack install` but I cannot load it:
> 
> ```
> bcoste@bbpv1:~/workspace/spack$ spack load subversion
> ModuleCmd_Load.c(213):ERROR:105: Unable to locate a modulefile for 'subversion/1.12.2'
> ```

The `subversion` module would probably still have to be whitelisted. Instructions are not supposed to be literal copy and paste ;)

**Comment by matz-e** [comment]
@wizmer I think the error occurs because we're trying to deploy 1.0.1 as per `deploy/packages/bbp-packages.yaml`. Please update the version there, too.

**Comment by wizmer** [comment]
Ok. I'm trying this. Thanks for your help.

**Comment by matz-e** [comment]
> Ok. I'm trying this. Thanks for your help.

You're welcome. Ideally, we'd catch this in the CI, but no-one has done a check for it so far…

**Review by matz-e — APPROVED** [review]
Since Jenkins swallowed this, I think we should be good to go. If there are any issues, it would probably be prudent to re-deploy `applications` into a new directory in April (this is more a note to @pramodk)

**Comment by pramodk** [comment]
> If there are any issues, it would probably be prudent to re-deploy applications into a new directory in April 

Why there will be any issues? I skimmed through comments and didn’t understand possible cause.

**Comment by matz-e** [comment]
> > If there are any issues, it would probably be prudent to re-deploy applications into a new directory in April
> 
> Why there will be any issues? I skimmed through comments and didn’t understand possible cause.

Maybe I'm just too cautious… there's probably nothing, but we delete a package here that may show up in a database. That's then always a little iffy. Hopefully not an issue, though.

**Pull request merged**